main.tex

```diff
@@ -448,7 +604,7 @@ \subsection{Credential Requests}
 For each $i \in [1, k]$ the user chooses an amount $a_i \mid 0 \leq a_i < a_{\mathit{max}}$ subject to the constraints of the balance proof (\cref{balance}). She commits to the amount with randomness $r_i \in_R \mathbb{Z}_q$, and these commitments are the attributes of the requested credentials:
 \[ M_{a_i}={G_h}^{r_i}{G_g}^{a_i} \]
 
-For each amount $a_i$ she also computes a range proof which ensures there are no negative values:
+For each amount $a_i$ she also computes a range proof which ensures the amounts are in the allowed range which ensures that finite field arithmetic will not overflow and the values will behave like positive integers when added or subtracted:
```

Comment:
> which ensures the amounts are in the allowed range which ensures

`which ensures` repeats twice in the same sentence.

Reply:
```suggestion
For each amount $a_i$ she also computes a proof that the amounts are in the allowed range, which ensures that finite field arithmetic will not overflow and the values will behave like positive integers when added or subtracted:
```